compiler/src/middle_end/matchcomp.re

```diff
@@ -900,6 +625,34 @@ let rec compile_matrix = mtx =>
           };
         Switch(ConstructorSwitch, switch_branches, default_tree);
       };
+    | ConstantMatrix =>
+      let constants = matrix_head_constants(mtx);
+      let equality_type = equality_type(List.hd(constants));
+
+      // TODO: Optimize physical equality checks into a switch.
```

Comment:
Do we want an issue for this?

Reply:
https://github.com/grain-lang/grain/issues/1185